tests/lib/helper.ts

```diff
@@ -1,15 +1,56 @@
 import { expect,Page } from '@playwright/test';
+import config from '../../tailwind.config.cjs'
+
+export const hexToRGB = (color?: string) => {
+  const hex = config.theme.extend.colors[color];
+
+  if (!hex) {
+    throw new Error(`hex not found for color: ${color}`)
+  }
+
+  let alpha = false;
+  let h = hex.slice(hex.startsWith('#') ? 1 : 0);
+  if (h.length === 3) h = [...h].map(x => x + x).join('');
+  else if (h.length === 8) alpha = true;
+  h = Number.parseInt(h, 16);
+  return (
+    `rgb${ 
+      alpha ? 'a' : '' 
+    }(${ 
+      h >>> (alpha ? 24 : 16) 
+    }, ${ 
+      (h & (alpha ? 0x00_FF_00_00 : 0x00_FF_00)) >>> (alpha ? 16 : 8) 
+    }, ${ 
+      (h & (alpha ? 0x00_00_FF_00 : 0x00_00_FF)) >>> (alpha ? 8 : 0) 
+    }${alpha ? `, ${h & 0x00_00_00_FF}` : '' 
+    })`
+  );
+};
 
 export const waitForCustomEvent = async(page:Page, customEventName:string) => {
-    return page.evaluate(eventName => new Promise(callback => window.addEventListener(eventName, callback,{ once: true })), customEventName)
+  return page.evaluate(eventName => new Promise(callback => window.addEventListener(eventName, callback,{ once: true })), customEventName)
 }
 
 export const waitForCustomEventTimeout = async (page:Page,customEventName:string) => {
-    const value = await Promise.race([
-        page.evaluate(eventName => new Promise(callback => window.addEventListener(eventName, callback, { once: true })), customEventName),
-        new Promise((resolve) => {setTimeout(resolve, 500, "timeout")}),
-    ])
+  const value = await Promise.race([
+    page.evaluate(eventName => new Promise(callback => window.addEventListener(eventName, callback, { once: true })), customEventName),
+    new Promise((resolve) => {setTimeout(resolve, 500, "timeout")}),
+  ])
 
-    expect(value).toBe("timeout")
+  expect(value).toBe("timeout")
 }
 
+// This function adds an event listener for eventName and puts the value of paramName on the window
+// Retrieve the value of paramName by calling await page.evaluate('window.<paramName>')
+export const waitForCustomEventWithParam = (page:Page, eventName:string, paramName:string) => {
```

Comment:
good stuff getting this to work. Are there any concerns with placing the value on the window causing issues between tests? is there any clean up we could do of the value afterwards so another test doesnt pick it up?

Reply:
Moving this comment to the other pr